app/packs/src/apps/mydb/elements/details/reactions/variationsTab/ReactionVariations.js

```diff
@@ -0,0 +1,231 @@
+/* eslint-disable react/display-name */
+import { AgGridReact } from 'ag-grid-react';
+import React, {
+  useRef, forwardRef, useState, useEffect, useImperativeHandle, useCallback
+} from 'react';
+import { v4 as uuidv4 } from 'uuid';
+import {
+  Button, FormGroup, Radio, ControlLabel, ButtonGroup,
+  OverlayTrigger, Tooltip, Form
+} from 'react-bootstrap';
+import _ from 'lodash';
+import { iupacNameTooltip } from 'src/apps/mydb/elements/details/reactions/schemeTab/Material';
+import { createVariationsRow } from 'src/apps/mydb/elements/details/reactions/variationsTab/ReactionVariationsUtils';
+
+function MaterialHeader({ material }) {
+  return (
+    <OverlayTrigger placement="bottom" overlay={iupacNameTooltip(material)}>
+      <div>
+        {material.external_label || material.short_label || material.id.toString()}
+      </div>
+    </OverlayTrigger>
+  );
+}
+
+function RowToolsCellRenderer({ data, copyRow, removeRow }) {
+  return (
+    <ButtonGroup>
+      <Button onClick={() => copyRow(data)}><i className="fa fa-copy" /></Button>
+      <Button onClick={() => removeRow(data)}><i className="fa fa-trash" /></Button>
+    </ButtonGroup>
+  );
+}
+
+function CellRenderer({ value: cellData }) {
+  const { value = '', unit = 'None', aux = {} } = cellData ?? {};
+  const cellContent = `${Number(value) ? Number(value).toFixed(3) : 'NaN'} [${unit}]`;
+
+  let overlayContent = aux.coefficient ? `Coeff: ${aux.coefficient}` : '';
+  overlayContent += aux.isReference ? '; Ref' : '';
+  overlayContent += aux.yield ? `; Yield: ${aux.yield}%` : '';
+  if (!overlayContent) {
+    return cellContent;
+  }
+
+  const overlay = (
+    <Tooltip>
+      {overlayContent}
+    </Tooltip>
+  );
+  return (
+    <OverlayTrigger placement="bottom" overlay={overlay}>
+      <div>
+        {cellContent}
+      </div>
+    </OverlayTrigger>
+  );
+}
+
+const CellEditor = forwardRef((props, ref) => {
+  const { value = '', unit = 'None', aux = {} } = props.value ?? {};
```

Comment:
Überschreibt er hier die Werte aus dem linken Objekt mit denen aus props.value? ich kenn die Form immer nur mit Keys, nicht zusätzlich mit values. Und bindet das = oder das ?? stärker?

Reply:
Genau, die Werte links werden mit denen aus `props.value` ueberschrieben insofern sie in `propes.value` definiert sind. Anderenfalls fallen sie auf die defaults (`''`, `'None'`, `{}`) zurueck (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#default_value).

Was meinst du mit staerkerer Bindung?

Reply:
wird zuerst { blakeks } = props.value ausgewertet und fällt wenn null auf {} zurück oder wird erst props.value ?? {} ausgewertet und dann { blakeks } = resultat bestimmt? (aka muss man sicherheitshalber klammern oder funktioniert das sauber so wie es soll)

Reply:
`props.value ?? {}` wird zuerst ausgefuehrt, da die operator precedence von `=` [right-to-left](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Operator_Precedence#table) ist.
D.h., `{ value = '', unit = 'None', aux = {} }` wird mit dem Resultat von `props.value ?? {}` ausgefuehrt.